File: clients/js/src/requests/item_names.ts
import { BaseRequest } from './base';

export type ItemNameRequest = BaseRequest<ItemNameRequestType, ItemNameRequestAttributes>;

enum ItemNameRequestType {
    item_names = 'item_names',
}

export interface ItemNameRequestAttributes {
    name: string;
}

export const ItemNameRequestDefault = (attributes: ItemNameRequestAttributes): ItemNameRequest => {
    return {
        data: {
            type: ItemNameRequestType.item_names,
            attributes,
        },
    };
};

File: pkmnapi-api/src/requests/item_names.rs
use rocket_okapi::JsonSchema;
use serde::Deserialize;

use crate::requests::base::BaseRequest;

pub type ItemNameRequest = BaseRequest<ItemNameRequestType, ItemNameRequestAttributes>;

impl ItemNameRequest {
    pub fn get_name(&self) -> &String {
        &self.data.attributes.name
    }
}

#[derive(Debug, Deserialize, JsonSchema)]
#[allow(non_camel_case_types)]
pub enum ItemNameRequestType {
    item_names,
}

#[derive(Debug, Deserialize, JsonSchema)]
pub struct ItemNameRequestAttributes {
    pub name: String,
}
